e2e/ui/Enterprise SSO/setup_link.spec.ts

```diff
@@ -0,0 +1,161 @@
+import { Locator, Page, expect, test as baseTest } from '@playwright/test';
+import { Portal } from 'e2e/support/fixtures';
+
+const TEST_SETUPLINK_REDIRECT_URL = 'http://localhost:3366';
+const TEST_SETUPLINK_DEFAULT_REDIRECT_URL = 'http://localhost:3366/login/saml';
+const TEST_SETUPLINK_MOCK_METADATA_URL = 'https://mocksaml.com/api/namespace/setuplink-test/saml/metadata';
+const TEST_SETUPLINK_URL_LABEL_SELECTOR =
+  'Share this link with your customers to allow them to set up the integrationClose';
+const TEST_SETUPLINK_ADMIN_URL = '/admin/sso-connection/setup-link';
+const TEST_SETUPLINK_ADMIN_CONNECTION = '/admin/sso-connection';
+
+type MyFixtures = {
+  portal: Portal;
+};
+
+export const test = baseTest.extend<MyFixtures>({
+  portal: async ({ page }, use) => {
+    const portal = new Portal(page);
+    await portal.doCredentialsLogin();
+    await use(portal);
+  },
+});
+
+async function retryForCondition(condition: Promise<any>, retries = 10, retryInterval = 500) {
+  while (retries) {
+    if (await condition) {
+      break;
+    } else {
+      setTimeout(() => {
+        retries -= 1;
+      }, retryInterval);
+    }
+  }
+}
+
+async function getRowByTenantProduct(
+  page: Page,
+  tenant: string,
+  product: string,
+  tenantProductIndexes: Array<number>
+): Promise<Locator | string> {
+  const rows = await page.getByRole('row').all();
+  const setupLinks = await Promise.all(rows.map(async (el: Locator) => el));
+  let row: Locator | string = '';
+
+  for (const link of setupLinks) {
+    const el = await link.innerText();
+    const cellContents = el.split('\t');
+    if (
+      cellContents.length > 1 &&
+      cellContents[tenantProductIndexes[0]] === tenant &&
+      cellContents[tenantProductIndexes[1]] === product
+    ) {
+      row = link;
+    }
+  }
+
+  return row;
+}
+
+test.describe('Admin Portal Enterprise SSO SetupLink using generic SAML 2.0', () => {
+  test('should be able to create setup link and sso connection using generic SAML 2.0', async ({ page }) => {
+    const tenant = 'acme-setuplink-test.com';
+    const product = 'acme-setuplink-test';
+
+    // Go to admin/sso-connection/setup-link page and create setup link
+    await page.goto(TEST_SETUPLINK_ADMIN_URL);
+    await page.getByRole('button', { name: 'New Setup Link' }).click();
+    await page.locator('input[type="text"]').nth(0).fill('acme-test');
+    await page.locator('input[type="text"]').nth(1).fill('acme test');
+    await page.locator('input[type="text"]').nth(2).fill(tenant);
+    await page.locator('input[type="text"]').nth(3).fill(product);
+    await page.locator('textarea').nth(0).fill(TEST_SETUPLINK_REDIRECT_URL);
+    await page.locator('input[type="url"]').nth(0).fill(TEST_SETUPLINK_DEFAULT_REDIRECT_URL);
+    await page.getByRole('button', { name: 'Create Setup Link' }).click();
+
+    // Extract generated setup link
+    await page.locator('input[type="text"]').first().click();
```

Comment:
See my previous comment ..
https://github.com/boxyhq/jackson/pull/2728/files/e5db9023704fa7acfaf2df536acfbd390eac0e33#r1610994403

Reply:
Also is this click necessary ? The next operation gets the inputValue from the textbox.